assembly/src/ast/parsers/context.rs

```diff
@@ -633,22 +639,23 @@ impl ParserContext<'_> {
     /// Return an error if the module with the specified name has not been imported via the `use`
     /// statement.
     fn get_imported_proc_id(
-        &self,
-        proc_name: &str,
+        &mut self,
+        proc_name: ProcedureName,
         module_name: &str,
         token: &Token,
     ) -> Result<ProcedureId, ParsingError> {
         let module_path = self
             .imports
             .get(module_name)
             .ok_or_else(|| ParsingError::procedure_module_not_imported(token, module_name))?;
-        let proc_id = ProcedureId::from_name(proc_name, module_path);
+        let proc_id = ProcedureId::from_name(proc_name.as_ref(), module_path);
+        self.used_imported_procs.insert(proc_id, (proc_name, module_path.clone()));
```

Comment:
Let's update the comments for this function to document this side effect.

Reply:
Fixed.